pkg/handlers/generic/mutation/imageregistries/credentials/credentials_secret.go

```diff
@@ -88,22 +86,53 @@ func kubeletStaticCredentialProviderSecretContents(configs []providerConfig) (st
 		RegistryHost string
 		Username     string
 		Password     string
+		Separator    string
 	}
 
-	inputs := make([]templateInput, 0, len(configs))
+	inputs := make([]templateInput, 0)
```

Comment:
nit: No need to pre-allocate now we don't know the necessary capacity
```suggestion
	var inputs []templateInput
```